### PR TITLE
Fix for #1232 causing cascading change detection disruption

### PR DESCRIPTION
**Pull request by hoffination**
added ngZone.run call to trigger change detection

**What kind of change does this PR introduce?** (check one with "x")
- [x ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
Change detection is not correctly firing when clicking on a row navigated to a new route, destroying the table. Some elements of the following view would not receive proper change detection. Triggering change detection manually would update the view, but was not triggered through the normal Angular lifecycle. This led us to believe there was an issue with zone.js.

This would only occur when routing from a virtualized row that had been rendered outside of the initial batch visible to the table. Example (render 10 visible rows, scroll down to 11th element and click, problems would occur). It would also occur when scrolling down and back up and clicking on one of the original elements. This led to the problem which was found in the ScrollerComponent: https://github.com/swimlane/ngx-datatable/blob/master/src/components/body/scroller.component.ts#L72

**What is the new behavior?**
Now change detection fires correctly on the subsequent route and the problem seems to have been resolved. Tests still pass and the application does not seem to have any slow down as a result.

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [ x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
Shoutout to @bennett000 and @andrewthauer for helping me find a fix to the problem!

### CODE REVIEW

File: src/components/body/scroller.component.ts [hunk 1]
```diff
@@ -69,7 +69,7 @@ export class ScrollerComponent implements OnInit, OnDestroy {
     requestAnimationFrame(() => {
```
Comment (bennett000) [inline src/components/body/scroller.component.ts]: @hoffination I know this `requestAnimationFrame` was here before, but I'm curious as to why?  It might be more advantageous to debounce this but it seems out of scope for the PR.  I wonder if the authors had an intention that isn't obvious to me?

Comment (hoffination) [reply src/components/body/scroller.component.ts]: Agreed. I'm not sure of the benefit

**Comment by amcdnl** [comment]
Thats a crazy case. Thanks for the detailed explanation.

However, I purposely don't want to put that calculation inside the zone. The zone kills performance, I want to take control of the whole lifecycle and CD manually for the scrolling case.

Just to be clear, this is just when you are clicking on a router link inside a cell?

File: src/components/body/scroller.component.ts (hunk 1 above)
Comment (amcdnl) [reply src/components/body/scroller.component.ts]: See: https://www.html5rocks.com/en/tutorials/speed/rendering/. The rows are positioned using css animations.

Comment (bennett000) [reply src/components/body/scroller.component.ts]: That makes sense, I usually debounce and I wondered if it was a vsync thing.

**Comment by NauxLiu** [comment]
@amcdnl I ran into a similar problem, listening to the `activate dblclick` event on the virtualized row that triggered the pop-up dialog box, which resulted in an exception.

**Comment by hoffination** [comment]
@amcdnl That makes sense. It happens when the `activate` callback is called on a row where we change the angular route from a component function. We filter the events down to just the click events. Not seeing any exceptions when this occurs.

**Comment by amcdnl** [comment]
So if thats the case, why not put it in the activate callback?

**Comment by hoffination** [comment]
@amcdnl Good suggestion! It looks like that resolved my issue in my project. I'd be happy to update the docs with a troubleshooting section including this solution if you can point me to the right place

**Comment by hoffination** [comment]
Also closing the PR as this isn't the solution we're moving forward with.
